homeassistant/components/onewire/__init__.py

```diff
@@ -26,10 +30,43 @@ async def async_setup_entry(hass: HomeAssistantType, config_entry: ConfigEntry):
 
     hass.data[DOMAIN][config_entry.unique_id] = onewirehub
 
-    for platform in PLATFORMS:
-        hass.async_create_task(
-            hass.config_entries.async_forward_entry_setup(config_entry, platform)
+    async def cleanup_registry() -> None:
+        # Get registries
+        device_registry, entity_registry = await asyncio.gather(
+            hass.helpers.device_registry.async_get_registry(),
+            hass.helpers.entity_registry.async_get_registry(),
         )
+        # Generate list of all device entries
+        registry_devices = [entry.id for entry in device_registry.devices.values()]
```

Comment:
We need to filter device entries for this config entry. Use `async_entries_for_config_entry`.